### PR TITLE
Cailey/artifactory/operator upgrade

### PR DESCRIPTION
**Pull request by caggles**

### CODE REVIEW

**Review by jefkel — CHANGES_REQUESTED** [review]
Looks like you caught the issue with the project roles?  Comments re: applying RBAC
- Cannot apply RBAC if the account applying the RBAC doesn't already have the same or better privileges.
- Cluster-roles for general access does not cover the expanded access that is needed within the namespace.

Changes Needed:
- Looks like you are deploying your operator image directly to production from your workstation.  I'd prefer to see the image pulled from the lab cluster (as that's our dev/test environment)
- I'd prefer to see the devops-artifactory.inv file either broken into different inventory files (for each deployment environment), or another approach to ensure we're maintaining both klab and silver environment parameters.

**Comment by caggles** [comment]
> Looks like you caught the issue with the project roles? Comments re: applying RBAC
> 
> * Cannot apply RBAC if the account applying the RBAC doesn't already have the same or better privileges.
> * Cluster-roles for general access does not cover the expanded access that is needed within the namespace.
> 
> Changes Needed:
> 
> * Looks like you are deploying your operator image directly to production from your workstation.  I'd prefer to see the image pulled from the lab cluster (as that's our dev/test environment)
> * I'd prefer to see the devops-artifactory.inv file either broken into different inventory files (for each deployment environment), or another approach to ensure we're maintaining both klab and silver environment parameters.

1. RBAC stuff has been caught and moved to the appropriate installation file.
2. Made a ticket to cover the pipeline stuff (#613)
3. Inventory files have been broken up into environment specific options.

File: apps/artifactory/artifactory-operator/README.md [hunk 1]
```diff
@@ -156,7 +58,7 @@ Example ArtifactorySA CR (Custom Resource) exists under `deploy/crds/tmpl-artifa
 Create the ArtifactorySA Custom Resource:
 
 ``` bash
- oc process -f ./deploy/crds/tmpl-artifactory-sa.yaml -p DESCRIPTOR="Description of Service Account" | oc create -f -
+ ocas process -f ./config/samples/tmpl-artifactory-sa.yaml -p NAME="test" -p DESCRIPTOR="Description of Service Account" | ocas create -f -
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-operator/README.md:61]: better specify ocas is an alias you set

File: apps/artifactory/artifactory-operator/config/default/manager_auth_proxy_patch.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# This patch inject a sidecar container which is a HTTP proxy for the
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-operator/config/default/manager_auth_proxy_patch.yaml:1]: are these configs static and for this version specific? is there need to parametrize them?

File: apps/artifactory/artifactory-operator/install/klab.inv [hunk 3]
```diff
@@ -0,0 +1,14 @@
+localhost
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-operator/install/klab.inv:1]: klab.inv? not .env?

**Review by ShellyXueHan — APPROVED** [review]

File: apps/artifactory/artifactory-operator/install/klab.inv (hunk 3 above)
Comment (caggles) [reply apps/artifactory/artifactory-operator/install/klab.inv:1]: yep, these are inventory files!

File: apps/artifactory/artifactory-operator/config/default/manager_auth_proxy_patch.yaml (hunk 2 above)
Comment (caggles) [reply apps/artifactory/artifactory-operator/config/default/manager_auth_proxy_patch.yaml:1]: this is actually part of the scaffolding of the new operator version, so i didn't write this specific bit

**Review by jefkel — APPROVED** [review]
My vote: Merge, test, remove this branch.. it's lived way too long now.

create new issues to resolve bugs that get introduced.  (Why are we reviewing long lived PR's with no scope/outcome descriptions?)

**Pull request merged**